tests/unit/test_charm_state.py

```diff
@@ -371,8 +371,14 @@ def test__parse_labels_invalid_labels(label_str: str, falsy_labels: tuple[str]):
         pytest.param(" a, b,   c", ("a", "b", "c"), id="comma separated labels with space"),
         pytest.param("1234", ("1234",), id="numeric label"),
         pytest.param("_", ("_",), id="underscore"),
+        pytest.param("-", ("-",), id="dash only"),
```

Comment:
Not sure if you have already checked, but we may need to check if this is actually an allowed label. But can be done in a follow-up PR.

Reply:
<img width="597" alt="image" src="https://github.com/canonical/github-runner-operator/assets/37652070/b81de7f0-02ee-45eb-9f82-cbc547ec79e6">
Github allows many of the labels that we don't allow - while I think flexibility is good, I think we should limit our offering to this set first so that we reduce security risks :)